app/src/main/java/org/projectbuendia/client/ui/dialogs/EditPatientDialogFragment.java

```diff
@@ -56,6 +62,8 @@
     @InjectView(R.id.patient_sex_female) RadioButton mSexFemale;
     @InjectView(R.id.patient_sex_male) RadioButton mSexMale;
 
+    private static final Pattern ID_PATTERN = Pattern.compile("([a-zA-Z]+)/?([0-9]+)*");
```

Comment:
Did you intend to replace the `+` with `*` instead of tacking it on at the end?

`([0-9]+)*` seems strange, and it looks like one of those regular expressions that can blow up the regex engine (runtime exponential in the length of the input, e.g.
 it will backtrack over the string `123456789123456789123456789x` trying something like 30 million different paths).